src/toplevel.c

```diff
@@ -782,7 +784,9 @@ void jl_check_static_parameter_conflicts(jl_lambda_info_t *li, jl_tuple_t *t, jl
             for(size_t j=0; j < nvars; j++) {
                 jl_value_t *tv = jl_tupleref(t,i);
                 if (jl_is_typevar(tv)) {
-                    if ((jl_sym_t*)jl_arrayref((jl_array_t*)jl_arrayref(vinfo,j),0) ==
+                    jl_value_t *vinfoj = jl_arrayref(vinfo,j);
+                    JL_GC_PUSH(&vinfoj);
+                    if ((jl_sym_t*)jl_arrayref((jl_array_t*)vinfoj,0) ==
```

Comment:
this is unnecessary, since the `jl_arrayref` will be non-allocating (looks like Jeff just decided to make that more obvious in https://github.com/JuliaLang/julia/commit/46f7d70fd75452fd0df99ca0aec1dd5177e60f99#diff-fd306b1b1448962b11c6840a936fdbc2R785)